runtime/compiler/control/JITServerCompilationThread.cpp

```diff
@@ -334,6 +336,7 @@ TR::CompilationInfoPerThreadRemote::waitForMyTurn(ClientSessionData *clientSessi
             !getWaitToBeNotified()) // Avoid a cohort of threads clearing the caches
             {
             clientSession->clearCaches();
+            compInfo->setNumClearedCaches(++numClearedCaches);
```

Comment:
This won't work because there could be a large time span between the moment you read the counter at the beginning of this function and the current moment where you increment it. These two operations need to stay close to one another. Better still, you can create a function that increments it: `incNumClearedCaches()`